### PR TITLE
feat(obstacle_cruise_planner): obstacle type dependent slow down for obstacle cruise planner

### PR DESCRIPTION
**Pull request by danielsanchezaran**
### <samp>🤖 Generated by Copilot at 82b40ad. Reformatted and expanded by Daniel</samp>

This pull request enhances the obstacle cruise planner to use the obstacle type information for planning --> if a set of parameters is defined for a given obstacle class, the  obstacle_cruise_planner will use those parameters to calculate the slow down speed. If no specific obstacle parameters are defined for the given obstacle type, the planner will use the "default" parameters instead. In the attached image there is an example that implements a different set of parameters for pedestrians only: 

image

Other changes: 
The PR adds the `object_classification` field to the `SlowDownObstacle` struct and uses it to select obstacle specific parameters for the slow down velocity calculation.

It also refactors the `SlowDownParam` struct and updates the `obstacle_cruise_planner.param.yaml` file to define parameters for different obstacle labels.

## Related links

Required launcher changes: [PR](https://github.com/autowarefoundation/autoware_launch/pull/621)

## Tests performed

Used the planning simulator to test changes in behavior for different obstacles. If the obstacle type matches a obstacle-specific parameter (defined in obstacle_cruise_planner.param.yaml), the planner uses those parameters to calculate the slow down speed (instead of the default parameters). 

See videos for BEFORE and AFTER for an example: [TIER IV restricted page](https://tier4.atlassian.net/browse/RT0-29332?atlOrigin=eyJpIjoiY2Y4MmI1NmRmNjRkNDU3Nzg4YzQzMmE1ZmE2NTIwODUiLCJwIjoiaiJ9)

I also modified the parameters while running the simulation to make sure the onParam changes execute properly. 

## Interface changes

obstacle_cruise_planner.param.yaml  has been changed. Now it is necessary to provide a "default" set of parameters for slow down and, if required, you can provide a new set of parameters for a specific obstacle type. See the image for an example change: 

image

## Effects on system behavior

it is possible to configure obstacle_cruise_planner to slow down differently depending on the obstacle type (ie. pedestrian, bicycle,car,etc.)

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].
- [ ] The PR has been properly tested.
- [ ] The PR has been reviewed by the code owners.

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.
- [ ] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: planning/obstacle_cruise_planner/include/obstacle_cruise_planner/planner_interface.hpp [hunk 1]
```diff
@@ -235,16 +266,37 @@ class PlannerInterface
         node.declare_parameter<double>("slow_down.lpf_gain_dist_to_slow_down");
     }
 
+    ObstacleSpecificParams getObstacleParamByLabel(const ObjectClassification & label_id) const
+    {
+      std::string label = types_map.at(label_id.label);
+      auto it = obstacle_to_param_struct_map.find(label);
+
+      if (it != obstacle_to_param_struct_map.end()) {
+        return obstacle_to_param_struct_map.at(label);
```
Comment (maxime-clem) [inline planning/obstacle_cruise_planner/include/obstacle_cruise_planner/planner_interface.hpp:275]:
```suggestion
      if (obstacle_to_param_struct_map.count(label) > 0) {
        return obstacle_to_param_struct_map.at(label);
```

File: planning/obstacle_cruise_planner/include/obstacle_cruise_planner/planner_interface.hpp [hunk 2]
```diff
@@ -235,16 +266,37 @@ class PlannerInterface
         node.declare_parameter<double>("slow_down.lpf_gain_dist_to_slow_down");
     }
 
+    ObstacleSpecificParams getObstacleParamByLabel(const ObjectClassification & label_id) const
+    {
+      std::string label = types_map.at(label_id.label);
```
Comment (maxime-clem) [inline planning/obstacle_cruise_planner/include/obstacle_cruise_planner/planner_interface.hpp:271]:
```suggestion
      const std::string label = types_map.at(label_id.label);
```

**Review by maxime-clem — COMMENTED** [review]
I did not test this PR but the code looks good.

File: planning/obstacle_cruise_planner/include/obstacle_cruise_planner/planner_interface.hpp (hunk 2 above)
Comment (danielsanchezaran) [reply planning/obstacle_cruise_planner/include/obstacle_cruise_planner/planner_interface.hpp]: Thanks for the suggestions @maxime-clem I have added them to the code.

**Review by kosuke55 — APPROVED** [review]
LGTM!

**Pull request merged**